pandas/core/algorithms.py

```diff
@@ -349,9 +354,10 @@ def factorize(values, sort=False, order=None, na_sentinel=-1, size_hint=None):
 
     table = hash_klass(size_hint or len(vals))
     uniques = vec_klass()
-    check_nulls = not is_integer_dtype(values)
-    labels = table.get_labels(vals, uniques, 0, na_sentinel, check_nulls)
+    check_nulls = (not is_integer_dtype(values)) and dropna
 
+    # the following line fills uniques:
```

Comment:
I added a comment because it took me a _very_ long time and some digging in the pyx files to understand what was going on. I am not expert enough to write a more general comment, but I would have liked to signal that to someone with the same knowledge of pandas as me, to avoid them the same difficulties I had

What should a more general comment look like?